src/imports.rs

```diff
@@ -105,7 +105,9 @@ impl Api for ExternalApi {
 
         let mut out = unsafe { consume_region(canon)? };
         out.truncate(read as usize);
-        Ok(CanonicalAddr(out))
+        // we require that the external implementation returns a byte slice containing base64-encoded value
+        // TODO: where to document this??
+        CanonicalAddr::from_external_base64(out)
```

Comment:
This should crash with real world canoncalize implementation (which tests don't have) because raw data is interpreted as a base64 string without encoding it.

Reply:
I assume the implementation changes to return Base64 encoded strings now...

I did update the mock encoding for tests. I can update the go encoding as well when we make bindings. Or I can bridge that in rust (accept `[]byte` and then just base64 encode it in go-cosmwasm).

This needs to be documented and updated when we update downstream repos